server/src/test/java/io/pravega/schemaregistry/service/SchemaRegistryServiceTest.java

```diff
@@ -707,4 +767,115 @@ public void testSchemaNormalization() {
         schema = service.getSchema(namespace, group, v.getId()).join();
         assertNotEquals(schema, secondOrder);
     }
+    
+    @Test
+    public void testSchemaInfoWithAndWithoutTypeAvro() {
+        SchemaStore schemaStore = SchemaStoreFactory.createInMemoryStore(executor);
+        SchemaRegistryService service = new SchemaRegistryService(schemaStore, executor);
+        String namespace = "n";
+        String group = "gav";
+        service.createGroup(namespace, group,
+                GroupProperties.builder().allowMultipleTypes(true).properties(ImmutableMap.<String, String>builder().build())
+                               .serializationFormat(SerializationFormat.Avro)
+                               .compatibility(Compatibility.allowAny()).build()).join();
+        
+        // case 1: empty type on record. Should take the name from the record
+        SchemaInfo withoutType = SchemaInfo.builder().type("").serializationFormat(SerializationFormat.Avro)
+                                      .schemaData(ByteBuffer.wrap(avroComplexObject.getBytes(Charsets.UTF_8)))
+                                      .properties(ImmutableMap.of()).build();
+        VersionInfo v = service.addSchema(namespace, group, withoutType).join();
+        assertEquals(v.getType(), "Person");
+        SchemaInfo schema = service.getSchema(namespace, group, v.getId()).join();
+        assertEquals(schema.getType(), "Person");
+        service.deleteSchema(namespace, group, v.getId());
+        
+        // case 2: empty name on union. this should take the name as union
```

Comment:
Please extract out these different cases into their own respective test cases/methods, so that each test case has a single responsibility.